src/puppetlabs/jdbc_util/core.clj

```diff
@@ -63,20 +64,30 @@
   ([result-set]
    (convert-result-arrays vec result-set))
   ([f result-set]
-   (let [convert #(cond
-                    (ks/array? %) (f %)
-                    (isa? (class %) java.sql.Array) (f (.getArray %))
-                    :else %)]
+   (let [convert (fn [v] (cond
+                           (ks/array? v) (f v)
+                           (instance? java.sql.Array v) (f (.getArray v))
+                           :else v))]
      (map #(ks/mapvals convert %) result-set))))
 
+(defn convert-result-pgobjects
+  "Converts PGObjects in a result set to be the value that they contain.
+  Values which aren't arrays are unchanged."
+  [result-set]
+  (let [val-if-pgobj (fn [v] (if (instance? PGobject v)
+                               (.getValue v)
+                               v))]
+    (map #(ks/mapvals val-if-pgobj %) result-set)))
+
 (defn query
   "An implementation of query that returns a fully evaluated result (no
-  JDBCArray objects, etc)"
+  lazy sequences, JDBCArray objects, or PGObjects)."
   [db sql-and-params]
   (let [convert (fn [rs]
                   (doall
-                    (convert-result-arrays vec
-                                           (jdbc/result-set-seq rs))))]
+                    (->> (jdbc/result-set-seq rs)
+                      (convert-result-arrays vec)
+                      (convert-result-pgobjects))))]
```

Comment:
if java.jdbc were reasonable there would be a way to register scoped handlers for different types of pgobject, but it's not